tools/crm_resource_runtime.c

```diff
@@ -188,18 +189,18 @@ find_matching_attr_resource(resource_t * rsc, const char * rsc_id, const char *
                 break;
         }
 
-    } else if (rsc->parent && BE_QUIET == FALSE) {
-        printf("Forcing %s of '%s' for '%s' instead of '%s'\n", cmd, attr_name, rsc_id, rsc->parent->id);
-
-    } else if(rsc->parent == NULL && rsc->children) {
-        resource_t *child = rsc->children->data;
+    } else {
+        for(GList *gIter = rsc->children; gIter; gIter = gIter->next) {
+            resource_t *child = gIter->data;
 
-        if(child->variant == pe_native) {
+            if(child->variant != pe_native) {
+                continue;
```

Comment:
If the resource is a cloned group, and the user gives the clone name, then the only child will be the group, so this will do nothing. If the user gives the group name instead, then it will hit the pe_clone case in the switch above.

Reply:
I refactored the whole function entirely. It has less conditions now. Basically, it returns all children for a resource. If the resource has a clone, it also returns the clone.